CMakeLists.txt

```diff
@@ -306,6 +306,33 @@ if(HAVE_NETINET_IP_H)
 	add_definitions(-DHAVE_NETINET_IP_H)
 endif()
 
+if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "(aarch64|arm64)")
+	set(HOST_AARCH64 true)
+elseif("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm")
+	set(HOST_ARM true)
+elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "SunOS" AND "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "i386")
+	set(HOST_X86_64 true)
+elseif("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "(x86_64|amd64)")
+	set(HOST_X86_64 true)
+elseif("${CMAKE_SYSTEM_NAME}" MATCHES "[i?86|x86]")
```

Comment:
It should check CMAKE_SYSTEM_PROCESSOR instead of CMAKE_SYSTEM_NAME, right?

Besides, the regex is wrong, maybe you are intended to write `(i?86|x86)`